pkg/build/webhook/generic/generic.go

```diff
@@ -66,6 +68,9 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 			glog.V(4).Infof("Error unmarshaling json %v, but continuing", err)
 			return nil, true, nil
 		}
+		if len(data.Env) > 0 {
+			updateBuildEnv(buildCfg, data.Env)
+		}
```

Comment:
On Tue, Apr 12, 2016 at 12:59 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/webhook/generic/generic.go
> <https://github.com/openshift/origin/pull/8477#discussion_r59412151>:
>
> > @@ -66,6 +68,9 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
> >  			glog.V(4).Infof("Error unmarshaling json %v, but continuing", err)
> >  			return nil, true, nil
> >  		}
> > +		if len(data.Env) > 0 {
> > +			updateBuildEnv(buildCfg, data.Env)
> > +		}
>
> I think there are a couple problems w/ this approach:
> 1) you're modifying the buildconfig whereas ideally we'd modify the build
> object
>
Well, ultimately the build object is modified, but yes, there is a middle
man, and it is not modified directly

> 2) you're modifying the buildconfig as a side-effect (it's not obvious
> from the function definition that the buildconfig which gets passed in is
> going to get mutated)
>
That's just a good point.  Simply missed that.

> I think you might want to take a look at the approach I used here:
> https://github.com/openshift/origin/pull/8405/files
>
If I read this right, you are saving information in the annotations to use
later.  If so, not 100% sure I like that approach.  That said, I think I
might have a tweak
to the Plugin's Extract method that might serve our purposes more
explicitly.  Let me work something up and we'll see what you think.

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8477/files/c6a3e2f5ebd4de6e829ec8a7d88ce568b2be7e80#r59412151>
>